File: event-bus/shared/src/main/java/com/github/dynamicextensionsalfresco/event/impl/Activator.java
package com.github.dynamicextensionsalfresco.event.impl;

import com.github.dynamicextensionsalfresco.event.EventBus;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;
import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

/**
 * @author Laurent Van der Linden
 */
public final class Activator implements BundleActivator {

    public void start(@NotNull BundleContext context) {
        Intrinsics.checkParameterIsNotNull(context, "context");
        context.registerService(EventBus.class, new DefaultEventBus(context), null);
    }

    public void stop(@NotNull BundleContext context) {
        Intrinsics.checkParameterIsNotNull(context, "context");
    }
}

File: event-bus/shared/src/main/kotlin/com/github/dynamicextensionsalfresco/event/impl/Activator.kt
package com.github.dynamicextensionsalfresco.event.impl

import com.github.dynamicextensionsalfresco.event.EventBus
import org.osgi.framework.BundleActivator
import org.osgi.framework.BundleContext

/**
 * @author Laurent Van der Linden
 */
public class Activator : BundleActivator {
    @Throws(Exception::class)
    override fun start(context: BundleContext) {
        context.registerService(EventBus::class.java, DefaultEventBus(context), null)
    }

    @Throws(Exception::class)
    override fun stop(context: BundleContext) {
    }
}
